### PR TITLE
implement android maxlength prop

### PR DESCRIPTION
**Pull request by nucleartux**
Fixes https://github.com/facebook/react-native/issues/3864
But I don't sure that this code is correct.

But it works and works same as in iOS.

### CODE REVIEW

File: Examples/UIExplorer/TextInputExample.android.js [hunk 1]
```diff
@@ -78,15 +78,25 @@ class RewriteExample extends React.Component {
     this.state = {text: ''};
   }
   render() {
+    var limit = 20;
+    var remainder = limit - this.state.text.length;
+    var remainderColor = remainder > 5 ? 'blue' : 'red';
```
Comment (nucleartux) [inline Examples/UIExplorer/TextInputExample.android.js]: I copy-pasted this from iOS example.

**Comment by nucleartux** [comment]
@ide @mkonicek @kmagiera please review. I can't wait for new features :)

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 2]
```diff
@@ -199,6 +200,15 @@ public void setNumLines(ReactEditText view, int numLines) {
     view.setLines(numLines);
   }
 
+  @ReactProp(name = "maxLength")
+  public void setMaxLength(ReactEditText view, @Nullable Integer maxLength) {
+    if(maxLength != null) {
```
Comment (kmagiera) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: Fix spacing:
```
if (maxLength != null) {
```

Also: you need to handle the case when length is null (property gets removed the component) - in that case you'd need to remove filers you set here

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 3]
```diff
@@ -199,6 +200,18 @@ public void setNumLines(ReactEditText view, int numLines) {
     view.setLines(numLines);
   }
 
+  @ReactProp(name = "maxLength")
+  public void setMaxLength(ReactEditText view, @Nullable Integer maxLength) {
+    if (maxLength == null) {
+      InputFilter[] filterArray = new InputFilter[0];
```
Comment (kmagiera) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: please extract this as private static constant (e.g. EMPTY_FILTERS) at the top of this class

**Pull request closed without merging**